### PR TITLE
docs: add react-hook-form examples in storybook

### PR DESCRIPTION
**Pull request by gui-santos**
# Purpose of PR

Integrates our Form Elements with react-hook-form and creates tests for them
Screenshot 2021-09-09 at 16 11 10

I will add documentation in the Website in another PR

## PR Checklist

- [ ] I have read the relevant `readme.md` file(s)
- [ ] All commits follow our [Git commit message convention](https://github.com/contentful/forma-36/tree/master/packages/forma-36-react-components#commits)
- [ ] Tests are added/updated/not required
- [ ] Tests are passing
- [ ] Storybook stories are added/updated/not required
- [ ] Usage notes are added/updated/not required
- [ ] Has been tested based on [Contentful's browser support](https://www.contentful.com/faq/about-contentful/#which-browsers-does-contentful-support)
- [ ] Doesn't contain any sensitive information

### CODE REVIEW

File: packages/components/forms/src/base-checkbox/BaseCheckboxGroup.tsx [hunk 1]
```diff
@@ -4,6 +4,10 @@ import type { CommonProps } from '@contentful/f36-core';
 import { BaseCheckboxGroupContext } from './BaseCheckboxGroupContext';
 
 export interface BaseCheckboxGroupProps extends CommonProps {
+  /**
+   * Handler that will be triggered when any checkbox inside the group loses focus
+   */
+  onBlur?: ChangeEventHandler<HTMLInputElement>;
```
Comment (massao) [inline packages/components/forms/src/base-checkbox/BaseCheckboxGroup.tsx:10]: Would also need to add to the import on line 1
```suggestion
  onBlur?: FocusEventHandler<HTMLInputElement>;
```

File: packages/components/forms/stories/ReactHookFormControlledInputs.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+            name="pears"
+            value="pears"
+          >
+            Pears
+          </ControlledCheckbox>
+
+          <ControlledCheckbox
+            control={control}
+            id="peaches"
+            name="peaches"
```
Comment (massao) [inline packages/components/forms/stories/ReactHookFormControlledInputs.stories.tsx:207]: Shouldn't all the checkboxes have the same name??

**Review by massao — COMMENTED** [review]
Should we also add a page with the code snippet on the f36-website??

File: packages/components/forms/src/base-checkbox/BaseCheckboxGroup.tsx (hunk 1 above)
Comment (gui-santos) [reply packages/components/forms/src/base-checkbox/BaseCheckboxGroup.tsx:10]: 🤦

File: packages/components/forms/stories/ReactHookFormControlledInputs.stories.tsx (hunk 2 above)
Comment (gui-santos) [reply packages/components/forms/stories/ReactHookFormControlledInputs.stories.tsx:207]: Yes! My bad!

**Comment by gui-santos** [comment]
> Should we also add a page with the code snippet on the f36-website??

@massao working on it right now in another branch

**Comment by denkristoffer** [comment]
Does this mean we are targeting this particular library?

**Comment by massao** [comment]
> Does this mean we are targeting this particular library?

Not really, we will also have an example using the Formik, the idea is to have an example on how to integrate with some well known third party libraries.

**Review by massao — APPROVED** [review]

**Pull request merged**